### PR TITLE
Fixes #RHIROS-1123 - check if container recommendation is valid

### PR DESCRIPTION
**Pull request by patilsuraj767**
Patch for - https://github.com/kruize/autotune/issues/770

### CODE REVIEW

File: internal/services/recommender.go [hunk 1]
```diff
@@ -54,7 +54,13 @@ func ProcessEvent(msg *kafka.Message) {
 		containers := data[0].Kubernetes_objects[0].Containers
 		for _, container := range containers {
 			for _, v := range container.Recommendations.Data {
-				if len(v.Duration_based.Short_term.Notifications) == 0 {
+
+				notifications := v.Duration_based.Short_term.Notifications
+
+				// Below "if" condition is the kruize patch for - https://github.com/kruize/autotune/issues/770
+				// issue - RHIROS-1123
+				// Should be removed once fixed in kruize
```
Comment (kgaikwad) [inline internal/services/recommender.go:62]:
```suggestion
				// Note - added patch here to handle kruize side issue
                                // Kruize issue: https://github.com/kruize/autotune/issues/770
                                // ROS issue - RHIROS-1123
                                // TODO: Should be removed once fixed in kruize
```

File: internal/services/recommender.go [hunk 2]
```diff
@@ -54,7 +54,13 @@ func ProcessEvent(msg *kafka.Message) {
...
+				if !(len(notifications) > 1 && notifications[0].Notificationtype == "info" && notifications[0].Message == "There is not enough data available to generate a recommendation.") {
```
Comment (kgaikwad) [inline internal/services/recommender.go:63]:
```suggestion
				if !(len(notifications) > 1 && notifications[0].Type == "info" && notifications[0].Message == "There is not enough data available to generate a recommendation.") {
```

Looking at `notifications` object from [here](https://github.com/kruize/autotune/issues/770), it says `type` not `notification_type`. 
OR is it something related to golang convention?

[the remark this answers is not in the document]
Comment (patilsuraj767) [reply internal/services/recommender.go:63]: We need `len(notifications) > 1` just to confirm notification array is not empty and we also need to verify the `Message`.  If needed we can skip the `Notificationtype` but I think it is good to have. 
Below is the exact json which we get. 
```
"notifications": [
                        {
                          "type": "info",
                          "message": "There is not enough data available to generate a recommendation."
                        }
                      ]

```

Comment (kgaikwad) [inline internal/services/recommender.go:63]: Can we declare any variable on top and make use of it here so that may be easy to locate? Thoughts?

**Review by kgaikwad — CHANGES_REQUESTED** [review]

File: internal/services/recommender.go (hunk 2 above)
Comment (patilsuraj767) [reply internal/services/recommender.go:63]: Yes, it is related to golang - [here](https://github.com/RedHatInsights/ros-ocp-backend/blob/main/internal/types/kruizePayload/common.go#L86)

Comment (patilsuraj767) [reply internal/services/recommender.go:63]: Are you asking for variable for string -` There is not enough data available to generate a recommendation.`?

[the remark this answers is not in the document]
Comment (patilsuraj767) [reply internal/services/recommender.go:63]: It should be `len(notifications) > 0`. I will make the changes.

**Comment by kgaikwad** [comment]
Thanks @patilsuraj767 for addressing all comments.

**Pull request merged**